src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -37,11 +37,38 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             var temps = ArrayBuilder<LocalSymbol>.GetInstance();
             var effects = DeconstructionSideEffects.GetInstance();
             ArrayBuilder<Binder.DeconstructionVariable> lhsTargets = GetAssignmentTargetsAndSideEffects(left, temps, effects.init);
+            BoundExpression result = RewriteDeconstruction(lhsTargets, conversion, left.Type, right, isUsed, temps, effects);
+            temps.Free();
+            effects.Free();
+            Binder.DeconstructionVariable.FreeDeconstructionVariables(lhsTargets);
+            return result;
+        }
+
+        private BoundExpression RewriteDeconstruction(
+            ArrayBuilder<Binder.DeconstructionVariable> lhsTargets,
+            Conversion conversion,
+            TypeSymbol leftType,
+            BoundExpression right,
+            bool isUsed,
+            ArrayBuilder<LocalSymbol> temps,
+            DeconstructionSideEffects effects)
+        {
+            if (right.Kind == BoundKind.ConditionalOperator)
+            {
+                var conditional = (BoundConditionalOperator)right;
+                Debug.Assert(!conditional.IsRef);
+                return conditional.Update(
+                    conditional.IsRef,
+                    VisitExpression(conditional.Condition),
+                    RewriteDeconstruction(lhsTargets, conversion, leftType, conditional.Consequence, isUsed: true, temps, effects),
+                    RewriteDeconstruction(lhsTargets, conversion, leftType, conditional.Alternative, isUsed: true, temps, effects),
+                    conditional.ConstantValue,
+                    leftType ?? conditional.Type);
```

Comment:
>leftType ?? conditional.Type [](start = 20, length = 28)

A deconstruction returns a value with the type of the left side. So I would expect this optimized case to also just have type `leftType` (always).
When is that not the case?